src/System.Globalization/tests/TextInfo.cs

```diff
@@ -0,0 +1,143 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Globalization;
+using Xunit;
+
+public class TextInfoTests
+{
+    [Theory]
+    [InlineData("")]
+    [InlineData("en-US")]
+    [InlineData("fr")]
+    [ActiveIssue(846, PlatformID.AnyUnix)]
+    public static void ToUpper(string localeName)
+    {
+        TextInfo ti = new CultureInfo(localeName).TextInfo;
+
+        Assert.Equal('A', ti.ToUpper('a'));
+        Assert.Equal("ABC", ti.ToUpper("abc"));
+        Assert.Equal('A', ti.ToUpper('A'));
+        Assert.Equal("ABC", ti.ToUpper("ABC"));
+
+        Assert.Equal("THIS IS A LONGER TEST CASE", ti.ToUpper("this is a longer test case"));
+        Assert.Equal("THIS IS A LONGER MIXED CASE TEST CASE", ti.ToUpper("this Is A LONGER mIXEd casE test case"));
+
+        Assert.Equal("THIS \t HAS \t SOME \t TABS", ti.ToUpper("this \t HaS \t somE \t TABS"));
+
+        Assert.Equal('1', ti.ToUpper('1'));
+        Assert.Equal("123", ti.ToUpper("123"));
+
+        Assert.Equal("EMBEDED\0NULL\0BYTE\0", ti.ToUpper("embeded\0NuLL\0Byte\0"));
+
+        Assert.Equal('\u00D3', ti.ToUpper('\u00F3'));
+        Assert.Equal("\u00D3", ti.ToUpper("\u00F3"));
+
+        Assert.Equal('\u2603', ti.ToUpper('\u2603'));
```

Comment:
Can do (the first is a character that actually cases (an o with an accent mark), the second is a character that does not (a snowman)).